test/unit/Cardano/Wallet/CoinSelection/RandomSpec.hs

```diff
@@ -0,0 +1,290 @@
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 100
+                , utxoInputs = [1,1,1,1]
+                , txOutputs = 2 :| [1]
+                , expectedResult = Right [1,1,1]
+                })
+        it "one output (targetMin=2), 3 medium inputs should select 1 coin" $ do
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 100
+                , utxoInputs = [5,5,5]
+                , txOutputs = 2 :| []
+                , expectedResult = Right [5]
+                })
+        it "two outputs (targetMin=2, targetMin=2), 3 big inputs should select 2 coins" $ do
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 100
+                , utxoInputs = [10,10,10]
+                , txOutputs = 2 :| [2]
+                , expectedResult = Right [10,10]
+                })
+        it "one output (targetMin=3), 6 small inputs should select 4 coins - maxNumOfInputs can cover targetMin but not targetAim" $ do
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 4
+                , utxoInputs = [1,1,1,1,1,1]
+                , txOutputs = 3 :| []
+                , expectedResult = Right [1,1,1,1]
+                })
+        it "one output (targetMin=3), 6 small inputs result in  - maxNumOfInputs cannot cover targetMin - fallback" $ do
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 2
+                , utxoInputs = [1,1,1,1,1,1]
+                , txOutputs = 3 :| []
+                , expectedResult = Left $ MaximumInputsReached 2
+                })
+        it "NotEnoughMoney error expected when not enough coins - fallback" $ do
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 100
+                , utxoInputs = [12,10,17]
+                , txOutputs = 40 :| []
+                , expectedResult = Left $ NotEnoughMoney 39 40
+                })
+        it "NotEnoughMoney error expected when not enough coins and utxo not fragmented enough - fallback" $ do
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 100
+                , utxoInputs = [12,10,17]
+                , txOutputs = 40 :| [1,1,1]
+                , expectedResult = Left $ NotEnoughMoney 39 43
+                })
+        it "UtxoNotEnoughFragmented error expected when enough coins and utxo not fragmented enough - fallback" $ do
+            (coinSelectionUnitTest
+                Fixture
+                { maxNumOfInputs = 100
+                , utxoInputs = [12,20,17]
+                , txOutputs = 40 :| [1,1,1]
+                , expectedResult = Left $ UtxoNotEnoughFragmented 3 4
+                })
+
+    describe "Coin selection properties : Random algorithm" $ do
+        xit "forall (UTxO, NonEmpty TxOut), \
```

Comment:
it stucks when `random` is called, probably because of wrong impl of MonadRandom instance of Identity at the end of this file